packages/transactional/src/lib/transactional.decorator.ts

```diff
@@ -113,20 +113,22 @@ export function Transactional(
                 `The @Transactional decorator can be only used on functions, but ${propertyKey.toString()} is not a function.`,
             );
         }
-        descriptor.value = function (...args: any[]) {
-            if (!this[transactionHostProperty]) {
-                throw new Error(
-                    `Failed to inject transaction host into ${target.constructor.name}`,
+        descriptor.value = new Proxy(original, {
+            apply: function (_, outerThis, ...args: any[]) {
```

Comment:
```suggestion
            apply: function (_, outerThis, args: any[]) {
```

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/apply), the signature of the apply trap is `apply(target, thisArg, argumentsList)`

I just realized that the tests were only passing, because the `@Transactional` decorator was only ever used on methods with a single argument that was later immediately stringified, which produces the same result:

```node
> (1).toString()
'1'
> ([1]).toString()
'1'
```

But this is wrong behavior and would break existing code.

Reply:
Totally right, completely overlooked it when doing the change.